packages/fast-components-react-base/src/tabs/tabs.tsx

```diff
@@ -0,0 +1,313 @@
+                ? this.props.activeId
+                : tabItems.length > 0
+                ? tabItems[0].props.id
+                : void(0)
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        const tabElements: JSX.Element[] = this.renderTabElements();
+
+        if (tabElements.length > 0) {
+            return (
+                <div
+                    {...this.unhandledProps()}
+                    className={this.generateClassNames()}
+                >
+                    <div
+                        role="tablist"
+                        ref={this.tabListRef}
+                        className={this.props.managedClasses.tab_list}
+                        aria-label={this.props.label}
+                        aria-orientation={this.props.orientation}
+                    >
+                        {tabElements}
+                    </div>
+                    {this.renderTabPanels()}
+                </div>
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * Generates class names based on props
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(get(this.props, "managedClasses.tabs"));
+    }
+
+    /**
+     * Renders the tab elements
+     */
+    private renderTabElements(): JSX.Element[] {
+        const tabElements: JSX.Element[] = [];
+
+        this.getChildBySlot(this.props.children, TabSlot.tabItem).forEach((tabItem: JSX.Element, index: number): void => {
+            const id: string = get(tabItem, "props.id") ? tabItem.props.id : `tab${index}`;
+
+            tabElements.push(
+                <button
+                    key={id}
+                    role="tab"
+                    className={this.generateTabClassNames(id)}
+                    aria-controls={id}
+                    aria-selected={this.state.activeId !== id ? false : true}
+                    onClick={this.handleClick}
+                    onKeyDown={this.handleKeyDown}
+                    tabIndex={this.state.activeId !== id ? -1 : 0}
+                >
+                    {this.getChildBySlot(tabItem.props.children, TabSlot.tab)[0]}
+                </button>
+            );
+        });
+
+        return tabElements;
+    }
+
+    /**
+     * Renders the tab panels
+     */
+    private renderTabPanels(): JSX.Element[] {
+        const tabPanels: JSX.Element[] = [];
+
+        this.getChildBySlot(this.props.children, TabSlot.tabItem).forEach((tabItem: JSX.Element, index: number): void => {
+            const id: string = get(tabItem, "props.id") ? tabItem.props.id : `tab${index}`;
+
+            tabPanels.push(
+                <div
+                    key={id}
+                    id={id}
+                    className={this.props.managedClasses.tab_panel}
+                    role="tabpanel"
+                    aria-labelledby={id}
+                    aria-hidden={this.state.activeId !== id ? true : false}
```

Comment:
nit: `aria-hidden={this.state.activeID !== id}`. The `true : false` is redundant

Reply:
oops, great point